ivy/functional/frontends/paddle/tensor/manipulation.py

```diff
@@ -148,3 +148,20 @@ def roll(x, shifts, axis=None, name=None):
 @to_ivy_arrays_and_back
 def take_along_axis(arr, indices, axis):
     return ivy.take_along_axis(arr, indices, axis)
+
+
+@with_supported_dtypes(
+    {
+        "2.5.1 and below": (
```

Comment:
> x (Tensor) – The input Tensor(or LoDTensor). The data type of the input Tensor x should be float16, float32, float64, int32, int64, bool. float16 is only supported on gpu.

Dtypes `bool` and `float16` are also supported on CPU and GPU respectively, please make the required changes in the decorator.